pkgs/native_assets_builder/doc/protocol/v1.md

```diff
@@ -0,0 +1,216 @@
+Build Hook Specification
+========================
+
+Version 1: As of 2024-01-24 implemented behind `--enable-experiment=native-assets`
+
+### Concepts
+
+#### Asset
+A file that is identified by an `assetId`. There may be multiple files with the
+same `assetId` with different characteristics such as `target` or `link_mode`.
+
+Note that in v1 there are _only_ native code assets, no data assets.
+
+#### AssetId
+An asset must have an `assetId`. Dart code that uses an asset, references the
+asset using the `assetId`.
+
+A package MUST prefix all assetIds it defines with: `package:<package>/`, this
+ensures assets don't conflict between packages.
+
+Conventionally, an asset referenced from `lib/src/foo.dart` in `package:foo`
+has `assetId = 'package:foo/src/foo.dart'`.
+
+#### Target
+
+A target specifies operating system and architecture.
+
+List of `platform` and `architecture` tuples supported by the
+Dart and Flutter SDKs to be specified below:
+ * `linux_x64`
+ * TODO: enumerate full list of supported platform/architecture tuples.
+
+
+
+### `build.dart`
+A package with native assets MUST define a `build.dart` file which accepts
```

Comment:
What is `A package with native assets` - if not a package which declares a `build.dart` file?